integration/src/main/java/scripts/testscripts/DatasetPermissionOperations.java

```diff
@@ -67,37 +77,82 @@ public void setup(List<TestUserSpecification> testUsers) throws Exception {
     adminTestSnapshotId = adminSnapshotsApi.createTestSnapshot(tdrDataset);
     tdrDatasetsApi.addDatasetPolicyMember(tdrDataset.getId(), "custodian", regularUser.userEmail);
     userTestSnapshotId = userSnapshotsApi.createTestSnapshot(tdrDataset);
-    adminTestDatasetId = adminCreateDataset(datasetRequestForSnapshot(adminTestSnapshotId));
+    adminTestSnapshotDatasetId =
+        adminCreateDataset(datasetRequest(adminTestSnapshotId.toString(), StorageSystem.TDR));
+
+    adminRawlsClient = new RawlsClient(server, adminUser);
+    userRawlsClient = new RawlsClient(server, regularUser);
+    adminTestWorkspace = adminRawlsClient.createTestWorkspace();
+    userTestWorkspace = userRawlsClient.createTestWorkspace();
+    adminTestWorkspaceDatasetId =
+        adminCreateDataset(datasetRequest(adminTestWorkspace.getWorkspaceId(), StorageSystem.WKS));
   }
 
   @Override
   public void userJourney(TestUserSpecification testUser) throws Exception {
-    testReaderPermissions();
-    testDiscovererPermissions();
-    testNoPermissions();
+    testReaderPermissionsForSnapshotDataset();
+    testDiscovererPermissionsForSnapshotDataset();
+    testNoPermissionsForSnapshotDataset();
     testAdminPermissionsOnUserSnapshot();
+
+    testReaderPermissionsForWorkspaceDataset();
+    testNoPermissionsForWorkspaceDataset();
+    testAdminPermissionsOnUserWorkspace();
   }
 
-  private void testReaderPermissions() throws Exception {
+  private void testReaderPermissionsForSnapshotDataset() throws Exception {
     setTestSnapshotPermissionForRegularUser("reader");
     // User cannot create a catalog entry on snapshot when user only has reader access on snapshot
-    CreateDatasetRequest request = datasetRequestForSnapshot(adminTestSnapshotId);
+    CreateDatasetRequest request =
+        datasetRequest(adminTestSnapshotId.toString(), StorageSystem.TDR);
     // note if this assertion fails we'll leave behind a stray dataset in the db
     assertThrows(ApiException.class, () -> userDatasetsApi.createDataset(request));
     assertThat(
         userDatasetsApi.getApiClient().getStatusCode(),
         is(HttpStatusCodes.STATUS_CODE_UNAUTHORIZED));
 
     // verify the user can still access data though
-    userDatasetsApi.getDataset(adminTestDatasetId);
+    userDatasetsApi.getDataset(adminTestSnapshotDatasetId);
     assertThat(userDatasetsApi.getApiClient().getStatusCode(), is(HttpStatusCodes.STATUS_CODE_OK));
   }
 
-  private void testDiscovererPermissions() throws Exception {
+  private void testReaderPermissionsForWorkspaceDataset() throws Exception {
+    setTestWorkspacePermissionForRegularUser(WorkspaceAccessLevel.READER);
+    // User cannot create a catalog entry on snapshot when user only has reader access on snapshot
+    CreateDatasetRequest request =
+        datasetRequest(adminTestWorkspace.getWorkspaceId(), StorageSystem.WKS);
```

Comment:
This is better, but would it be possible to make a version that could be used for both storage systems? What if this method accepted a `datasetRequest` argument and a `adminTestDatasetId` argument?